### DISCUSSION TITLE
pest vs tree-sitter

### DISCUSSION THREAD

**Discussion by perehonchuk** [pest-parser/pest#1054] [General]
what is the difference between pest and tree-sitter? Where would you use tree-sitter instead of pest and vice-versa?

**Comment by tomtau** [comment]
tree-sitter is based on GLR, while pest is based on PEG; tree-sitter is more general / expressive, can do incremental parsing etc., so IMHO it's great for processing of programming languages (e.g. for IDEs where one need to re-parse the same files frequently). pest is simpler, so it's great for one-off processing of textual inputs where regular expressions aren't enough to capture their complexity (e.g. processing of custom configuration formats or DSLs)
